docs/config_entries_config_flow_handler.md

```diff
@@ -276,7 +276,14 @@ import voluptuous as vol
 class ExampleConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Config flow for Example integration."""
 
-    async def async_step_reconfigure(self, user_input: dict[str, Any] | None = None):
+    async def async_step_reconfigure(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        return await async_step_reconfigure_confirm()
```

Comment:
Add type hints, and split into two steps to highlight that the initial input contains the data from the config entry.